### ISSUE TITLE
`input_fn.list` implementation for custom estimator is broken

### ISSUE THREAD

**Issue by terrytangyuan** [rstudio/tfestimators#107]
Looks like [this commit](https://github.com/rstudio/tfestimators/commit/b25ed6ba13b50657f30c3362d5f0493b2e547b95#diff-9ae668830f25aa9f78614de86f263bd5) breaks `input_fn.list` for custom estimators. 

Note that we were using `TRUE` and `FALSE` to toggle different `input_fn` implementations, which was not correct after that commit because we are [checking `estimator`'s class to toggle the different implementations](https://github.com/rstudio/tfestimators/blob/master/R/input_fn.R#L180-L186). I just changed the test cases for `input_fn` to use `estimator` instead here: https://github.com/rstudio/tfestimators/commit/bf7107b41e01e224dfe10e0ccfdb3695dee15bdb. 

Then I found [this particular test case](https://github.com/rstudio/tfestimators/blob/master/tests/testthat/test-input-fn.R#L94-L104) is now broken after the commit that tries to consolidate the implementations. 

@kevinushey Could you help look into this when you get a chance?

**Comment by kevinushey** [comment]
Sorry to come back to this again, but do we _really_ need to support this? In my experience, users almost never encounter or use data in this 'list-of-lists' format; it's almost always tabular. The only exception in the R universe is so-called list-columns, but those are still used as members of a data frame.

Do we have some evidence of real, live TensorFlow applications that expect data in this format?

**Comment by jjallaire** [comment]
I think text data is often in this form factor. Also, the TF Datasets API
and documentation seem to imply that this is an anticipated/mainstream use
case (they have lots of stuff for dealing with nested structures of tensors
as "datasets"

**Comment by jjallaire** [comment]
I agree however that if the list structure is fundamentally just defining a rectangular structure expressed as a nested list that we are fine just asking users to get this into a data frame or matrix. Let's see what @terrytangyuan has to say here.

**Comment by terrytangyuan** [comment]
Main motivation comes from need to feed in sequences with variable length to RNNs ([this post might be helpful](https://danijar.com/variable-sequence-lengths-in-tensorflow/)) without padding as part of the pre-processing (thus might be non-rectangular structure). Example datasets are irregular time series, sequence data for translation tasks, etc. 

That being said, we can also probably hold off on this until we have RNN estimators or time series estimators in the future.

**Comment by dfalbel** [comment]
Also some models deals with list of inputs. See for example ACGAN here: https://github.com/rstudio/keras/blob/master/vignettes/examples/mnist_acgan.R#L242
